modules/gapi/test/infer/gapi_infer_ie_test.cpp

```diff
@@ -75,6 +75,44 @@ void normAssert(cv::InputArray ref, cv::InputArray test,
     EXPECT_LE(normInf, lInf) << comment;
 }
 
+std::vector<std::string> modelPathByName(const std::string &model_name) {
+    // Handle OMZ model layout changes among OpenVINO versions here
+    static const std::unordered_multimap<std::string, std::string> map = {
+        {"age-gender-recognition-retail-0013",
+         "intel/age-gender-recognition-retail-0013/FP32"},
+        {"age-gender-recognition-retail-0013",
+         "Retail/object_attributes/age_gender/dldt/age-gender-recognition-retail-0013"},
+    };
+    const auto range = map.equal_range(model_name);
+    std::vector<std::string> result;
+    for (auto it = range.first; it != range.second; ++it) {
+        result.emplace_back(it->second);
+    }
+    return result;
+}
+
+std::tuple<std::string, std::string> findModel(const std::string &model_name) {
+    const auto candidates = modelPathByName(model_name);
+    CV_Assert(!candidates.empty() && "No model path candidates found at all");
+
+    for (auto &&path : candidates) {
+        std::string model_xml, model_bin;
+        try {
+            model_xml = findDataFile(path + "/" + model_name + ".xml", false);
+            model_bin = findDataFile(path + "/" + model_name + ".bin", false);
```

Comment:
This should be fixed:
```.diff
-Retail/object_attributes/age_gender/dldt/age-gender-recognition-retail-0013/age-gender-recognition-retail-0013.xml
+Retail/object_attributes/age_gender/dldt/age-gender-recognition-retail-0013.xml
```

BTW, On build share format of these models is IRv5 (from OpenVINO 2018R5 AFAIK) which is deprecated. Perhaps we should migrate on new variant (from new version of OpenVINO/OMZ).

Reply:
Fixing this... thanks! Updated our wiki too.

Reply:
@alalek 

Model lookup works now but it couldn't find grace hopper pic. \('o')/

```
[ RUN      ] ROIList.TestInfer
unknown file: Failure
C++ exception with description "OpenCV(4.4.0-pre) /build/precommit_custom_mac/opencv/modules/ts/src/ts.cpp:1052: error: (-2:Unspecified error) OpenCV tests: Can't find required data file: grace_hopper_227.png in function 'findData'
" thrown in the test fixture's constructor.
[  FAILED  ] ROIList.TestInfer (2 ms)
```

Reply:
Because this image belongs to "dnn".
Use "../dnn/grace_hopper_227.png" instead (gapi tests has non-existed "gapi" base, see `test_main.cpp`).

Reply:
> Because this image belongs to "dnn".

And dnn is added to lookup path, that's the problem. I have these tests running locally.

Reply:
```c++
        if (dnnDataPath) {
            // Add the dnnDataPath itself - G-API is using some images there directly
            cvtest::addDataSearchPath(dnnDataPath);
```

Reply:
`"../dnn/grace_hopper_227.png"` worked in one of three custom builds now. :)

Reply:
gave up & dropped this, random data ftw.